circus-train-gcp/src/main/java/com/hotels/bdp/circustrain/gcp/CredentialProviderRelativePathFactory.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.gcp;
+
+import java.nio.file.Paths;
+
+import com.hotels.bdp.circustrain.gcp.context.GCPSecurity;
+
```

Comment:
This class could probably have a better name, maybe simply `GCPCredentialProvider` the factory is implementation details, then you could just pass the `GCPCredentialProvider` to the `GCPCredentialCopier` and `@Autowire` this class using it as a replacement for the GCPSecurity POJO, and do similar for the `HdfsGsCredentialAbsolutePathFactory`